file-attribute-caching/src/main/kotlin/com/pkware/file/fileAttributeCaching/ExpirableCache.kt

```diff
@@ -0,0 +1,97 @@
+package com.pkware.file.fileAttributeCaching
+
+import java.time.Clock
+import java.util.function.Function
+
+/**
+ * A generic cache that expires after [flushInterval] milliseconds.
```

Comment:
I am not yet comfortable with the idea of having a cache whose expiration is determined by a fixed time. By that, I mean I expect either Java or the native file system should already be taking care of this for us.

Reply:
Well the cache time definitely should not be changing from run to run. Marius agreed with me that 5 seconds was ideal.

Reply:
okay we are getting rid of the cache for now and timing and just adopting fields for each attribute type

I would still like to do testing with both